drivers/gpio/gpio_nrfx.c

```diff
@@ -119,6 +122,7 @@ static int gpio_nrfx_pin_configure(const struct device *port, gpio_pin_t pin,
 	uint8_t ch;
 	bool free_ch = false;
 	const struct gpio_nrfx_cfg *cfg = get_port_cfg(port);
+	struct gpio_nrfx_data *data = get_port_data(port);
```

Comment:
```suggestion
#ifdef CONFIG_GPIO_GET_CONFIG
	struct gpio_nrfx_data *data = get_port_data(port);
#endif
```
Otherwise data is unused